Yank/repex.py

```diff
@@ -1859,7 +1859,8 @@ def run(self, n_iterations=None):
             if (self._online_analysis_interval is not None and
                     self._iteration > self._online_analysis_minimum_iterations and
                     self._iteration % self._online_analysis_interval == 0):
-                self._run_online_analysis()
+                # Executed only by node 0 and broadcasted to be used in _is_completed().
+                self._last_err_free_energy = self._run_online_analysis()
```

Comment:
Doesn't this assignment redundantly set the `self._last_err_free_energy`? or is it something to do with how MPI handles information?

Reply:
> or is it something to do with how MPI handles information?

Yes, exactly. Only node 0 calculates `_last_err_free_energy`, and then it broadcasts its value to the other processes so that they can check independently for completion.